src/SourceConstraints.hs

```diff
@@ -179,6 +180,20 @@ sortedIEs dynFlags ies =
       Module . render dynFlags $ unLoc moduleName
     ieClass ie = error $ "Unsupported: " <> gshow ie
 
+sortedIEThingWith :: DynFlags -> IE GhcPs -> Maybe SDoc
+sortedIEThingWith dynFlags (IEThingWith _xIE _wrappedName _ieWildcard ieWith _ieFieldLabels) =
+  if rendered /= expected
+     then pure $ text . message $ intercalate ", " expected
+     else empty
+  where
+    message :: String -> String
+    message example = "Unsorted import/export with list: (" <> example <> ")"
+
+    rendered = render dynFlags <$> ieWith
+    expected = sort rendered
+
+sortedIEThingWith _ _ = empty
```

Comment:
@mbj Wouldn't we typically still provide (`_` prefixed) names even when they're unused?

Reply:
IMO only if we'd otherwise not have named them in an adjacent pattern.